### PR TITLE
bump gcp driver and sidecar versions

### PR DESCRIPTION
**Pull request by RomanBednar**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
This is a followup of previous version bump discussed [here](https://github.com/kubernetes/kubernetes/pull/110105#issuecomment-1135969096) to allow gcp driver tests to use latest snapshotter version.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
 "NONE"
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/testing-manifests/storage-csi/gce-pd/controller_ss.yaml [hunk 1]
```diff
@@ -39,7 +39,7 @@ spec:
             - name: socket-dir
               mountPath: /csi
         - name: csi-provisioner
-          image: registry.k8s.io/sig-storage/csi-provisioner:v3.1.0
+          image: registry.k8s.io/sig-storage/csi-provisioner:v3.2.0
```
Comment (xing-yang) [inline test/e2e/testing-manifests/storage-csi/gce-pd/controller_ss.yaml:42]: 3.2.1 is the latest

File: test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpath-plugin.yaml [hunk 2]
```diff
@@ -323,7 +323,7 @@ spec:
             name: socket-dir
 
         - name: csi-provisioner
-          image: registry.k8s.io/sig-storage/csi-provisioner:v3.1.0
+          image: registry.k8s.io/sig-storage/csi-provisioner:v3.2.0
```
Comment (xing-yang) [inline test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpath-plugin.yaml:326]: 3.2.1

File: test/e2e/testing-manifests/storage-csi/mock/csi-mock-driver.yaml [hunk 3]
```diff
@@ -15,7 +15,7 @@ spec:
       serviceAccountName: csi-mock
       containers:
         - name: csi-provisioner
-          image: registry.k8s.io/sig-storage/csi-provisioner:v3.0.0
+          image: registry.k8s.io/sig-storage/csi-provisioner:v3.2.0
```
Comment (xing-yang) [inline test/e2e/testing-manifests/storage-csi/mock/csi-mock-driver.yaml:18]: 3.2.1

File: test/e2e/testing-manifests/storage-csi/mock/csi-mock-proxy.yaml [hunk 4]
```diff
@@ -15,7 +15,7 @@ spec:
       serviceAccountName: csi-mock
       containers:
         - name: csi-provisioner
-          image: registry.k8s.io/sig-storage/csi-provisioner:v3.0.0
+          image: registry.k8s.io/sig-storage/csi-provisioner:v3.2.0
```
Comment (xing-yang) [inline test/e2e/testing-manifests/storage-csi/mock/csi-mock-proxy.yaml:18]: 3.2.1

**Comment by xing-yang** [comment]
There's another PR that tries to update the images: https://github.com/kubernetes/kubernetes/pull/110826

**Comment by RomanBednar** [comment]
> There's another PR that tries to update the images: #110826

I see, thanks. Closing this one.

**Pull request closed without merging**